systemtest/src/test/java/io/strimzi/systemtest/oauth/OauthST.java

```diff
@@ -0,0 +1,217 @@
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.strimzi.api.kafka.model.CertSecretSourceBuilder;
+import io.strimzi.systemtest.Constants;
+import io.strimzi.systemtest.MessagingBaseST;
+import io.strimzi.systemtest.utils.StUtils;
+import io.strimzi.test.executor.Exec;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.codec.BodyCodec;
+import io.vertx.junit5.VertxExtension;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static io.strimzi.systemtest.Constants.NODEPORT_SUPPORTED;
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import static io.strimzi.systemtest.Resources.deployKeycloakNodePortService;
+
+@Tag(NODEPORT_SUPPORTED)
+@Tag(REGRESSION)
+@ExtendWith(VertxExtension.class)
+public class OauthST extends MessagingBaseST {
+
+    public static final String NAMESPACE = "oauth2-cluster-test";
+    protected static final Logger LOGGER = LogManager.getLogger(OauthST.class);
+
+    protected static final String TOPIC_NAME = "my-topic";
+    static final String REVERSE_TOPIC_NAME = "my-topic-reversed";
+
+    static final String PRODUCER_USER_NAME = "hello-world-producer";
+    static final String CONSUMER_USER_NAME = "hello-world-consumer";
+    static final String STREAMS_USER_NAME = "hello-world-streams";
+
+    private static final String PRODUCER_OAUTH_SECRET = "hello-world-producer-oauth";
+    private static final String CONSUMER_OAUTH_SECRET = "hello-world-consumer-oauth";
+    private static final String STREAMS_OAUTH_SECRET = "hello-world-streams-oauth";
+    static final String CONNECT_OAUTH_SECRET = "my-connect-oauth";
+    static final String MIRROR_MAKER_OAUTH_SECRET = "my-mirror-maker-oauth";
+    static final String BRIDGE_OAUTH_SECRET = "my-bridge-oauth";
+    static final String OAUTH_KEY = "clientSecret";
+
+    static String oauthTokenEndpointUri;
+    static String validIssuerUri;
+    static String jwksEndpointUri;
+    static String userNameClaim;
+    static final String CERTIFICATE_OF_KEYCLOAK = "tls.crt";
+    static final String SECRET_OF_KEYCLOAK = "x509-https-secret";
+
+    static String clusterHost = "";
+    static final String BRIDGE_EXTERNAL_SERVICE = CLUSTER_NAME + "-bridge-external-service";
+    protected WebClient client;
+
+    JsonObject sendHttpRequests(JsonObject records, String bridgeHost, int bridgePort) throws InterruptedException, ExecutionException, TimeoutException {
```

Comment:
Same method is already in `HttpBridgeBaseST`, maybe you can refactor things and re-use it?

Reply:
done. I moved all methods with http to our `HttpUtils`.